src/middleware/basic-auth/index.ts

```diff
@@ -15,11 +15,13 @@ type BasicAuthOptions =
       password: string
       realm?: string
       hashFunction?: Function
+      invalidUserMessage?: string | object | Function
```

Comment:
It's better to write the more specific type like this instead of `Function`: 

```ts
type MessageFunction = (c: Context) => string | object | Promise<string | object>
```

Reply:
Yes that's a good idea. PR is updated.